homeassistant/components/deconz/button.py

```diff
@@ -57,34 +57,21 @@ async def async_setup_entry(
     gateway.entities[DOMAIN] = set()
 
     @callback
-    def async_add_scene(scenes: list[PydeconzScene] | None = None) -> None:
+    def async_add_scene(_: EventType, scene_id: str) -> None:
         """Add scene button from deCONZ."""
-        entities = []
-
-        if scenes is None:
-            scenes = list(gateway.api.scenes.values())
-
-        for scene in scenes:
-
-            known_entities = set(gateway.entities[DOMAIN])
-            for description in ENTITY_DESCRIPTIONS.get(PydeconzScene, []):
-
-                new_entity = DeconzButton(scene, gateway, description)
-                if new_entity.unique_id not in known_entities:
-                    entities.append(new_entity)
-
-        if entities:
-            async_add_entities(entities)
+        scene = gateway.api.scenes[scene_id]
+        for description in ENTITY_DESCRIPTIONS.get(PydeconzScene, []):
+            async_add_entities([DeconzButton(scene, gateway, description)])
```

Comment:
```suggestion
        async_add_entities(
            DeconzButton(scene, gateway, description)
            for description in ENTITY_DESCRIPTIONS.get(PydeconzScene, [])
        )
```

This should be a bit more efficient

Reply:
Thanks @bdraco